examples/pytorch/torchscript/IrisClassification/iris_classification.py

```diff
@@ -0,0 +1,113 @@
+        self.fc1 = nn.Linear(4, 10)
+        self.fc2 = nn.Linear(10, 10)
+        self.fc3 = nn.Linear(10, 3)
+
+    def forward(self, x):
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = F.dropout(x, 0.2)
+        x = self.fc3(x)
+        return x
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def prepare_data():
+    iris = load_iris()
+    data = iris.data
+    labels = iris.target
+    target_names = iris.target_names
+
+    X_train, X_test, y_train, y_test = train_test_split(
+        data, labels, test_size=0.2, random_state=42, shuffle=True, stratify=labels
+    )
+
+    X_train = torch.FloatTensor(X_train).to(device)
+    X_test = torch.FloatTensor(X_test).to(device)
+    y_train = torch.LongTensor(y_train).to(device)
+    y_test = torch.LongTensor(y_test).to(device)
+
+    return X_train, X_test, y_train, y_test, target_names
+
+
+def train_model(model, epochs, X_train, y_train):
+    criterion = nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
+
+    for epoch in range(epochs):
+        out = model(X_train)
+        loss = criterion(out, y_train).to(device)
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+
+        if epoch % 10 == 0:
+            print("number of epoch", epoch, "loss", float(loss))
+
+    return model
+
+
+def test_model(model, X_test, y_test):
+    model.eval()
+    with torch.no_grad():
+        predict_out = model(X_test)
+        _, predict_y = torch.max(predict_out, 1)
+
+        print(
+            "\nprediction accuracy", float(accuracy_score(y_test.cpu(), predict_y.cpu())),
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Iris Classification Torchscripted model")
+
+    parser.add_argument(
+        "--tracking-uri", type=str, default="http://localhost:5000/", help="MLflow tracking uri"
+    )
+    parser.add_argument(
+        "--epochs", type=int, default=100, help="number of epochs to run (default: 100)"
+    )
+
+    args = parser.parse_args()
+
+    model = IrisClassifier()
+    model = model.to(device)
+    X_train, X_test, y_train, y_test, target_names = prepare_data()
+    scripted_model = torch.jit.script(model)  # scripting the model
+    scripted_model = train_model(scripted_model, args.epochs, X_train, y_train)
+    test_model(scripted_model, X_test, y_test)
+
+    mlflow.tracking.set_tracking_uri(args.tracking_uri)
+    with mlflow.start_run() as run:
+        mlflow.pytorch.log_model(scripted_model, "model")  # logging scripted model
+        model_path = mlflow.get_artifact_uri("model")
+        loaded_pytorch_model = mlflow.pytorch.load_model(model_path)  # loading scripted model
+        model.eval()
+        with torch.no_grad():
+            test_datapoint = pd.DataFrame([[4.4000, 3.0000, 1.3000, 0.2000]])
+            actual = "setosa"
+            pywrap = _PyTorchWrapper(loaded_pytorch_model)
```

Comment:
`_PyTorchWrapper` is not meant to be used like this. You can use `mlfow.pyfunc.load_model` instead which returns a model wrapped by `_PyTorchWrapper`.

Reply:
Okie. Loading the scripted model and making prediction is already part of Unit Test - https://github.com/mlflow/mlflow/blob/fa1676e3156b8e5adfb957e00230691b5cb179e5/tests/pytorch/test_pytorch_model_export.py#L307 . Shall i revert to the old code where we make the prediction or you want me to update the code and make predictions based on pyfunc model ?

Reply:
I was expecting something like:

```python
loaded_pytorch_model = mlflow.pytorch.load_model(model_path)
preds = loaded_pytorch_model(X)

print(preds)
```

Reply:
Thats how i had it earlier - https://github.com/mlflow/mlflow/pull/3557/commits/62690812d7dca3b7566936681373a3cc7a82b281 . will revert this commit.